api/src/service/domain/workflow/subproject_projected_budget_delete.spec.ts

```diff
@@ -0,0 +1,274 @@
+import { assert } from "chai";
+
+import { Ctx } from "../../../lib/ctx";
+import * as Result from "../../../result";
+import { BusinessEvent } from "../business_event";
+import { NotAuthorized } from "../errors/not_authorized";
+import { NotFound } from "../errors/not_found";
+import { ServiceUser } from "../organization/service_user";
+import { Subproject } from "./subproject";
+import { deleteProjectedBudget } from "./subproject_projected_budget_delete";
+
+const ctx: Ctx = { requestId: "", source: "test" };
+const root: ServiceUser = { id: "root", groups: [] };
+const alice: ServiceUser = { id: "alice", groups: ["alice_and_bob", "alice_and_bob_and_charlie"] };
+const bob: ServiceUser = { id: "bob", groups: ["alice_and_bob", "alice_and_bob_and_charlie"] };
+const charlie: ServiceUser = { id: "charlie", groups: ["alice_and_bob_and_charlie"] };
+const projectId = "dummy-project";
+const subprojectId = "dummy-subproject";
+const baseSubproject: Subproject = {
+  id: subprojectId,
+  projectId,
+  createdAt: new Date().toISOString(),
+  status: "open",
+  displayName: "dummy",
+  description: "dummy",
+  currency: "EUR",
+  projectedBudgets: [],
+  workflowitemOrdering: [],
+  permissions: { "subproject.budget.deleteProjected": [alice, bob, charlie].map(x => x.id) },
+  log: [],
+  additionalData: {},
+};
+const baseRepository = {
+  getUsersForIdentity: async identity => {
+    if (identity === "alice") return ["alice"];
+    if (identity === "bob") return ["bob"];
+    if (identity === "charlie") return ["charlie"];
+    if (identity === "alice_and_bob") return ["alice", "bob"];
+    if (identity === "alice_and_bob_and_charlie") return ["alice", "bob", "charlie"];
+    if (identity === "root") return ["root"];
+    throw Error(`unexpected identity: ${identity}`);
+  },
+};
+
+describe("Update subproject projected budget: permissions", () => {
+  it("Without the subproject.budget.updateProjected permission, a user cannot update a projected budget.", async () => {
+    const result = await deleteProjectedBudget(
+      ctx,
+      alice,
+      projectId,
+      subprojectId,
+      "Othertestcorp",
+      "EUR",
+      {
+        ...baseRepository,
+        getSubproject: async () => ({ ...baseSubproject, permissions: {} }),
+      },
+    );
+
+    // NotAuthorized error due to the missing permissions:
+    assert.isTrue(Result.isErr(result));
+    assert.instanceOf(result, NotAuthorized);
+  });
+
+  it("The root user doesn't need permission to update a projected budget", async () => {
+    const result = await deleteProjectedBudget(
+      ctx,
+      root,
+      projectId,
+      subprojectId,
+      "Othertestcorp",
```

Comment:
The id's don't match with the existing project budget

Reply:
Fixed